dbt-snowflake/src/dbt/include/snowflake/macros/metadata/list_relations_without_caching.sql

```diff
@@ -12,26 +12,32 @@
         dbt is currently configured to list a maximum of {{ max_results_per_iter * max_iter }} objects per schema.
         {{ schema }} exceeds this limit. If this is expected, you may configure this limit
         by setting list_relations_per_page and list_relations_page_limit in your project flags.
-        It is recommended to start by increasing list_relations_page_limit to something more than the default of 10.
+        It is recommended to start by increasing list_relations_page_limit.
     {%- endset -%}
 
-    {%- set paginated_results = [] -%}
-    {%- set watermark = none -%}
+    {%- set paginated_state = namespace(paginated_results=[], watermark=none) -%}
 
     {%- do run_query('alter session set quoted_identifiers_ignore_case = false;') -%}
 
-    {#- loop an extra time to catch the breach of max iterations -#}
-    {%- for _ in range(0, max_iter + 1) -%}
-
-        {#- raise an error if we still didn't exit and we're beyond the max iterations limit -#}
-        {%- if loop.index == max_iter -%}
-            {%- do exceptions.raise_compiler_error(too_many_relations_msg) -%}
+    {#-
+        loop an extra time to catch the breach of max iterations
+        Note: while range is 0-based, loop.index starts at 1
+    -#}
+    {%- for _ in range(max_iter + 1) -%}
+
+        {#-
+            raise a warning and break if we still didn't exit and we're beyond the max iterations limit
+            Note: while range is 0-based, loop.index starts at 1
+        -#}
+        {%- if loop.index == max_iter + 1 -%}
+            {%- do exceptions.warn(too_many_relations_msg) -%}
+            {%- break -%}
         {%- endif -%}
 
-        {%- set show_objects_sql = snowflake__show_objects_sql(schema, max_results_per_iter, watermark) -%}
+        {%- set show_objects_sql = snowflake__show_objects_sql(schema, max_results_per_iter, paginated_state.watermark) -%}
         {%- set paginated_result = run_query(show_objects_sql) -%}
-        {%- do paginated_results.append(paginated_result) -%}
-        {%- set watermark = paginated_result.columns.get('name').values()[-1] -%}
+        {%- do paginated_state.paginated_results.append(paginated_result) -%}
+        {%- set paginated_state.watermark = paginated_result.columns[1].values()[-1] -%}
```

Comment:
why get by index instead of key?

Reply:
Honestly, I made this change when troubleshooting because this is how we used to fetch this column. I just never reverted back. I'll fix it.